k-distribution/src/main/scripts/lib/pyk/kastManip.py

```diff
@@ -206,22 +321,28 @@ def _collapseDots(_kast):
                 newArgs.append(ktokenDots)
             return KApply(label, newArgs)
         elif isKRewrite(_kast):
-            if _kast["lhs"] == ktokenDots:
+            if _kast['lhs'] == ktokenDots:
                 return ktokenDots
         return _kast
     return traverseBottomUp(kast, _collapseDots)
 
 def pushDownRewrites(kast):
+    """Traverse a term and push rewrites down as far as possible.
+
+    -   Input: Kast term potentially with rewrites.
+    -   Output: Kast term with rewrites localized (or removed) as much as possible.
+    """
     def _pushDownRewrites(_kast):
         if isKRewrite(_kast):
-            lhs = _kast["lhs"]
-            rhs = _kast["rhs"]
+            lhs = _kast['lhs']
+            rhs = _kast['rhs']
             if lhs == rhs:
                 return lhs
-            if  isKApply(lhs) and isKApply(rhs) and lhs["label"] == rhs["label"] and isCellKLabel(lhs["label"]) \
-            and len(lhs["args"]) == len(rhs["args"]):
-                    newArgs = [ KRewrite(lArg, rArg) for (lArg, rArg) in zip(lhs["args"], rhs["args"]) ]
-                    return KApply(lhs["label"], newArgs)
+            if isKVariable(lhs) and isKVariable(rhs) and lhs['name'] == rhs['name']:
+                return lhs
+            if  isKApply(lhs) and isKApply(rhs) and lhs['label'] == rhs['label'] and len(lhs['args']) == len(rhs['args']):
+                    newArgs = [ KRewrite(lArg, rArg) for (lArg, rArg) in zip(lhs['args'], rhs['args']) ]
+                    return KApply(lhs['label'], newArgs)
```

Comment:
Is this correctly indented? Seems a bit off from the rest.

Reply:
Fixed.